### PR TITLE
KVM: Enable SSL if keystore exists

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR enables SSL in kvm agent if keystore exists.

This fixes the issue in URL check of direct-download templates

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]
LGTM

File: agent/src/main/java/com/cloud/agent/AgentShell.java [hunk 1]
```diff
@@ -397,6 +399,24 @@ public void init(String[] args) throws ConfigurationException {
...
+    private void enableSSL() {
+        final File agentFile = PropertiesUtil.findConfigFile("agent.properties");
+        if (agentFile == null) {
+            s_logger.info("Failed to find agent.properties file");
+            return;
+        }
+        String keystorePass = getProperty(null, "keystore.passphrase");
+        if (StringUtils.isBlank(keystorePass)) {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply agent/src/main/java/com/cloud/agent/AgentShell.java:411]: @DaanHoogland no, each keystore has non-empty passphrase

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test ubuntu20 kvm-ubuntu20

**Comment by weizhouapache** [comment]
@blueorangutan test ubuntu20 kvm-ubuntu20

**Comment by weizhouapache** [comment]
@blueorangutan test ubuntu20 kvm-ubuntu20

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**